### PR TITLE
fix(orc8r): Fix code scanning alert - Disabled TLS certificate check …

### LINKED ISSUE #10405: Fix code scanning alert - Disabled TLS certificate check
Tracking issue for:
- [ ] https://github.com/magma/magma/security/code-scanning/10

### PR DESCRIPTION
**Pull request by nikolakn**
…#10405

Signed-off-by: Nikola Knezevic <nkcodeplus@gmail.com>

fix(orc8r): Fix code scanning alert - Disabled TLS certificate check

## Summary

adding constant in config instead of using it directly in code

### CODE REVIEW

**Comment by hcgatewood** [comment]
cc @emakeev for review on how this affects his golang agw

**Review by hcgatewood — APPROVED** [review]
Lgtm, but please wait to merge till we hear from @emakeev

**Comment by themarwhal** [comment]
Removing the ready2merge label until we hear from @emakeev ? (see Hunter's comment above)

File: orc8r/gateway/go/config/control_proxy_config.go [hunk 1]
```diff
@@ -46,27 +47,29 @@ type ControlProxyCfg struct {
...
+		RootCaFile:                DefaultRootCaFile,
+		GwCertFile:                DefaultGwCertFile,
+		GwCertKeyFile:             DefaultGwCertKeyFile,
+		LocalPort:                 DefaultLocalPort,
+		CloudAddr:                 "",
+		CloudPort:                 DefaultCloudPort,
+		BootstrapAddr:             "",
+		BootstrapPort:             DefaultBootstrapPort,
+		ProxyCloudConnection:      DefaultProxyCloudConnection,
+		CloudConnectionSkipVerify: DefaultSkipVerify,
```
Comment (emakeev) [inline orc8r/gateway/go/config/control_proxy_config.go:72]: CloudBootstrapperSkipVerify: ...

File: orc8r/gateway/go/services/bootstrapper/service/rpc.go [hunk 2]
```diff
@@ -101,7 +102,7 @@ func (b *Bootstrapper) getGrpcOpts(useProxy bool, cfg *config.ControlProxyCfg) [
 			}
 		} else {
 			tlsCfg = &tls.Config{
-				InsecureSkipVerify: true,
+				InsecureSkipVerify: cfg.CloudConnectionSkipVerify,
```
Comment (emakeev) [inline orc8r/gateway/go/services/bootstrapper/service/rpc.go:105]: func (b *Bootstrapper) getGrpcOpts(useProxy bool, cfg *config.ControlProxyCfg) ([]grpc.DialOption, error) {
...
...
...
} else if cfg.CloudBootstrapperSkipVerify {
    tlsCfg = &tls.Config{
				InsecureSkipVerify: true ,
    }
} else {
    return opts, fmt.Errorf("no root CA to verify server authenticity")
}
...

**Review by emakeev — COMMENTED** [review]
The rpc.go is only using insecure cloud connection if there are neither Root CA nor OS cert chain available. And it only uses this connection for bootstrapping.

I'd call the new flag/option 'CloudConnectionSkipVerify' that you added something like: 'AllowInsecureBootstrap' or 'CloudBootstrapperSkipVerify' to avoid confusion and the logic should probably change to erroring out immediately if there are no root CA/OS CA chain and insecure cloud connection is not allowed (please see inline comments) it would prevent an attempt to connect that is going to fail anyway.

At a minimum, I'd rename the option, erroring out immediately is an optimization and is mostly nice to have.

**Pull request closed without merging**